win32ss/gdi/dib/dib16bpp.c

```diff
@@ -139,19 +140,90 @@ DIB_16BPP_VLine(SURFOBJ *SurfObj, LONG x, LONG y1, LONG y2, ULONG c)
 BOOLEAN
 DIB_16BPP_BitBltSrcCopy(PBLTINFO BltInfo)
 {
-  LONG     i, j, sx, sy, xColor, f1;
+  LONG     i, j, sx, sy, xColor, f1, lTmp;
   PBYTE    SourceBits, DestBits, SourceLine, DestLine;
   PBYTE    SourceBits_4BPP, SourceLine_4BPP;
+  BOOLEAN  bTopToBottom, bLeftToRight;
+
+  DPRINT("DIB_16BPP_BitBltSrcCopy: SrcSurf cx/cy (%d/%d), DestSuft cx/cy (%d/%d) dstRect: (%d,%d)-(%d,%d)\n",
+    BltInfo->SourceSurface->sizlBitmap.cx, BltInfo->SourceSurface->sizlBitmap.cy,
+    BltInfo->DestSurface->sizlBitmap.cx, BltInfo->DestSurface->sizlBitmap.cy,
+    BltInfo->DestRect.left, BltInfo->DestRect.top, BltInfo->DestRect.right, BltInfo->DestRect.bottom);
+
+  /* If we came from copybits.c with a TBltInfo->SourceSurface->fjBitmap & BMF_TOPDOWN   */
+  /* bit set, then we need a flip of bTopToBottom. This mostly fixes Lazarus and PeaZip. */
+
+  DPRINT("SourceSurface->fjBitmap & BMF_TOPDOWN is '%d'.\n", BltInfo->SourceSurface->fjBitmap & BMF_TOPDOWN);
+
+  /* Get back flip here */
+  if (BltInfo->DestRect.left > BltInfo->DestRect.right)
+  {
+    bLeftToRight = TRUE;
+  }
+  else
+  {
+    bLeftToRight = FALSE;
+  }
+
+  /* The OR for BltInfo->SourceSurface->fjBitmap & BMF_TOPDOWN checks for coming from copybits.c */ 
+  if ((BltInfo->DestRect.top > BltInfo->DestRect.bottom) || (BltInfo->SourceSurface->fjBitmap & BMF_TOPDOWN))
+  {
+    bTopToBottom = TRUE;
+  }
+  else
+  {
+    bTopToBottom = FALSE;
+  }
+
+  DPRINT("BltInfo->SourcePoint.x is '%d' & BltInfo->SourcePoint.y is '%d'.\n",
+    BltInfo->SourcePoint.x, BltInfo->SourcePoint.y);
+
+  /* Make WellOrdered with top < bottom and left < right */
+  if (BltInfo->DestRect.left > BltInfo->DestRect.right)
+  {
+    lTmp = BltInfo->DestRect.left;
+    BltInfo->DestRect.left = BltInfo->DestRect.right;
+    BltInfo->DestRect.right = lTmp;
+  }
+
+  if (BltInfo->DestRect.top > BltInfo->DestRect.bottom)
+  {
+    lTmp = BltInfo->DestRect.top;
+    BltInfo->DestRect.top = BltInfo->DestRect.bottom;
+    BltInfo->DestRect.bottom = lTmp;
+  }
```

Comment:
Can we use `RECTL_vMakeWellOrdered` ?

Reply:
Yes, this can be converted to ```RECTL_vMakeWellOrdered(&BltInfo->DestRect);```

Reply:
Conversions done in multiple locations. Thanks.